internal/explorer/helpers.go

```diff
@@ -287,21 +290,58 @@ func (a *App) getNodeData(nodeID string, force bool) (string, error) {
 			}
 			return "", ErrNodeNotFound
 		} else if err != nil {
-			err = a.DeleteRedisKey(fmt.Sprintf("GRID3NODE:%s", nodeID))
-			if err != nil {
-				log.Warn().Err(err).Msg("could not delete key in redis")
+
+			redisData := NodeInfo{}
+			json.Unmarshal([]byte(value), &redisData)
+
+			// up or loading make it likely down for 10 minutes
+			if redisData.Status == "up" {
+				// mark the node likely down if we can't reach this node in 10 mins it's down
+				nodeInfo.Status = "likely down"
+				marshalledInfo, err := json.Marshal(nodeInfo)
```

Comment:
Instead of always calling json.Marshal, and json.Unmarshal the nodeInfo type can has the methods to serialize, and deserialize which will make it easier (and cleaner) to change the encoding for example and make it more abstract